src/blsprivatekey.cpp

```diff
@@ -220,21 +220,35 @@ std::vector<uint8_t> BLSPrivateKey::Serialize() const {
     return data;
 }
 
-BLSSignature BLSPrivateKey::Sign(const uint8_t *msg, size_t len) const {
+BLSInsecureSignature BLSPrivateKey::SignInsecure(const uint8_t *msg, size_t len) const {
```

Comment:
I don't like this. I prefer having a method in BLSSignature to convert it to an insecure signature. So you can just do BLSPrivateKey::Sign(...).ToInsecure()

Reply:
That would bring us back to unnecessary bookkeeping of aggregation info when the caller doesn't need it, which was a big problem for me tbh. If done how you suggested it, it would temporarily create a BLSSignature with proper aggregation info and then throw it away.

The idea behind BLSInsecureSignature is that it is the actual internal implementation of plain signatures. As you can see in BLSSignature, verification is handed over to BLSInsecureSignature after the proper exponentiation has been performed.

I understand that you don't like having 2 different APIs here. That's why I chose to explicitly mark all low-level/insecure methods as insecure. These aren't even documented in the README.md, so a not-knowing developer wouldn't even encounter the insecure API when following the examples. If a developer encounters these insecure methods and uses them, he should be aware of what he's doing (we're warning him in the name of every method he calls...). If he still uses them, he either knows what he's doing or he probably shouldn't work on crypto projects (I know, harsh, but that's my humble opinion) because someone ignoring such obvious warning would tend to do other huge mistakes as well.

Reply:
OK, you convinced me